### ISSUE TITLE
handle device orientation / rotation

### ISSUE THREAD

**Issue by RandomEtc** [cooperhewitt/Planetary#69]
I added some basic code for this at our own app layer, but there are some extra things we need to do to make the status bar appear on the correct side when the home button is double-tapped: http://stackoverflow.com/questions/5357457/ipad-opengl-screen-rotation-problems-multitasking-bar-and-popups-not-rotating

We also need to adapt all our UI classes and label code to render at the current orientation.

**Comment by RandomEtc** [comment]
OK, here's the fix for the multi-tasking status bar orientation, placed in `AppCocoaTouch.mm` in `setupCocoaTouchWindow` before `[app->mState->mCinderView startAnimation]`:

```
// make sure the task bar is shown on the correct side of the device
[[NSNotificationCenter defaultCenter] addObserverForName:@"UIDeviceOrientationDidChangeNotification" object:nil queue:nil usingBlock: ^(NSNotification *notification) {
    UIDeviceOrientation orientation = [[UIDevice currentDevice] orientation];
    if (UIDeviceOrientationIsValidInterfaceOrientation(orientation)) {
        [UIApplication sharedApplication].statusBarOrientation = UIInterfaceOrientation(orientation);
    }
}];    
[[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
```

Will post to the forum later in case anyone needs it - not sure if it has any undesired side-effects as-is but it could also be the basis of dispatching an event for orientation instead of deriving from the accelerometer.

**Comment by RandomEtc** [comment]
moving to deviceRotation branch of bloomtime/Kepler and handling unknown rotations. closed by 633227518f929a65604838fba01dd832d822272d.
